src/jobserver_test.cc

```diff
@@ -0,0 +1,357 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "jobserver.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include <cassert>
+#include <cstdlib>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+/// Wrapper class to provide access to protected members of the Jobserver class.
+struct JobserverTest : public Jobserver {
+  /// Forwards calls to the protected Jobserver::ParseJobserverAuth() method.
+  bool ParseJobserverAuth(const char* type) {
+    return Jobserver::ParseJobserverAuth(type);
+  }
+
+  /// Provides access to the protected Jobserver::jobserver_name_ member.
+  const char* JobserverName() const {
```

Comment:
stylistic nit: it helps when method names are verbs that describe an action, as this avoids ambiguities. For methods that return simple values such as this one, it is clearer to use something like  `GetJobserverName()`. A different style is to use lower-cased method names for methods that just return a direct value / property, e.g. `jobserver_name()`.

Reply:
I went with `GetJobserverName()`